identity/randomid.go

```diff
@@ -46,5 +50,6 @@ func NewID() string {
 
 	var nn big.Int
 	nn.SetBytes(p[:])
-	return fmt.Sprintf("%0[1]*s", maxRandomIDLength, nn.Text(randomIDBase))
+	base36 := fmt.Sprintf("%0[1]*s", maxRandomIDLength+1, nn.Text(randomIDBase))
+	return base36[1 : maxRandomIDLength+1]
```

Comment:
This can simply be:

```
return nn.Text(randomIDBase)[:maxRandomIDLength]
```

Since we always generate extra bytes, we no longer need to have padding from `fmt.Sprintf`.